landoapi/hgexports.py

```diff
@@ -319,6 +319,13 @@ def parse_email_body(self, content: str) -> tuple[str, str]:
                 commit_message_lines += [""]
 
             commit_message_lines.append(line)
+        else:
```

Comment:
Are there any more specific/explicit conditions we could check here?

Reply:
Unfortunately I don't think so. The format essentially has subject/date/author etc information in the Unix Mbox style headers, and then the message body. The message body looks like:

- the extended commit message if one is available (ie a commit message longer than 1 line).
- a `---` line to signify the end of the commit message.
- the `--stat` output from Git (ie like you would find for `git log --stat`).
- a line beginning with `diff --git` that signifies the start of the diff, followed by all the lines of the diff.
- the final two lines of the patch body are `--` on one line, followed by the Git version used to generate the patch on the last line.

If the commit is empty, the `---` isn't present to signal the end of the commit message, and only the `--` + Git version is found. So I think it makes sense to consider commits where we don't find `---`  as empty commits.